### ISSUE TITLE
Replacing public drawables

### ISSUE THREAD

**Issue by Falseclock** [rovo89/XposedBridge#49]
Have public drawable resource in res/values/public.xml of SystemUI.apk

``` xml
    <public type="drawable" name="stat_notify_alarm" id="0x7f02034e" />
```

on JB&KK I always used following code to replace drawables inside handleInitPackageResources() like that:

``` java
resparam.res.setReplacement("com.android.systemui", "drawable", "stat_notify_alarm", modRes.fwd(R.drawable.new_stat_notify_alarm));
```

but on Lollipop this code does not work. It does not show any exceptions, but also do nothing.

If I'm trying to replace by ID

``` java
int targetID = resparam.res.getIdentifier("stat_notify_alarm" "drawable", "com.android.systemui");
resparam.res.setReplacement(targetID, modRes.fwd(R.drawable.new_stat_notify_alarm));
```

I'm getting exception "Object has already been initialized" from initObject of XResources.java

What I'm doing wrong?

**Comment by rovo89** [comment]
Assuming that you create your modRes correctly, your code looks right. Resource replacement in general is working, so I have to try and reproduce this. Help from others is of course appreciated while I take care of low-level fixes...

**Comment by pylerSM** [comment]
Do you have latest Xposed build?

**Comment by Falseclock** [comment]
@rovo89 ,

``` java
    public static String MODULE_PATH;

    public void initZygote(StartupParam startupParam) throws Throwable
    {
        MODULE_PATH = startupParam.modulePath;
    }

    public void handleInitPackageResources(InitPackageResourcesParam resparam) throws Throwable
    {
        SystemUI.handleColoredWiFi(resparam, MODULE_PATH);
    }

    ........
    // SystemUI.class 
    public static void handleColoredWiFi(InitPackageResourcesParam resparam, String path)
    {
        final XModuleResources modRes = XModuleResources.createInstance(path, resparam.res);
        resparam.res.setReplacement("com.android.systemui", "drawable", "stat_sys_wifi_signal_inandout_1", modRes.fwd(R.drawable.stat_sys_wifi_signal_inandout_1));
    }
```

@pylerSM, yes, Alpha 4 installed.

**Comment by Falseclock** [comment]
And yes, this happens only for "public" drawables declared in res/values/public.xml. In this case no any record inside 

``` java
public final class R {
    public static final class drawable {

    }
}
```

**Comment by rovo89** [comment]
I will try to reproduce it when I find the time for it. But are you sure that this drawable is actually used? If it's not in R.drawable, that might be an indicator that it's obsolete. At least in AOSP, it was removed a long time ago: https://android.googlesource.com/platform/frameworks/base.git/+/fdfbbf62e61020b6b6170862eaabd72b43bca4ea

**Comment by tonyp** [comment]
I did use `resparam.res.setReplacement` with alpha4 to replace a stat_sys_\* drawable on AOSP 5.0 and it worked fine. I switched to a fakeID implementation as it offers more flexibility, but in theory it should work I believe.

It might be interesting to know what ROM you're using, in case it has some conflicting theming engine or something similiar?
As rovo mentioned, the `stat_notify_alarm` drawable is called `stat_sys_alarm` at AOSP now. But as you used another drawable in your example (`stat_sys_wifi_signal_inandout_1`) which isn't part of AOSP I assume you're using a stock ROM - a quick Google Search brings up HTC.
